ocaml/xapi/pvs_proxy_control.ml

```diff
@@ -100,7 +100,18 @@ let update_site_on_localhost ~__context ~site ~vdi ?(starting_proxies=[]) ?(stop
 	let proxies = starting_proxies @ (List.set_difference local_running_proxies stopping_proxies) |> List.setify in
 	let proxy_config = metadata_of_site ~__context ~site ~vdi ~proxies in
 	if proxy_config.clients <> [] then
-		Network.Net.PVS_proxy.configure_site dbg proxy_config
+		try Network.Net.PVS_proxy.configure_site dbg proxy_config
+		with e ->
+			let (vif, proxy) = List.hd starting_proxies in
```

Comment:
This `List.hd` will lead to an bad exception when `starting_proxies = []`, which is the case when it is called from `stop_proxy`. I think that it is better to send the alert from the caller, i.e. `start_proxy` below, because there you know exactly which proxy you need to report about.